src/overlays/actors/ovl_En_Test7/z_en_test7.c

```diff
@@ -32,80 +46,943 @@ const ActorInit En_Test7_InitVars = {
+        } else if ((globalCtx->nextEntranceIndex == 0x9A80) && (gSaveContext.weekEventReg[33] & 0x80)) {
+            globalCtx->nextEntranceIndex = 0xAE80;
+        }
+    }
+
+    globalCtx->sceneLoadFlag = 0x14;
+    globalCtx->unk_1887F = 2;
+    gSaveContext.seqIndex = 0xFF;
+    gSaveContext.nightSeqIndex = 0xFF;
+}
+
+void func_80AF24D8(EnTest7* this, GlobalContext* globalCtx, f32 arg2) {
+    Vec3f sp3C;
+    Vec3f* pos;
+    Player* player = GET_PLAYER(globalCtx);
+    Camera* camera = Play_GetCamera(globalCtx, ActorCutscene_GetCurrentCamera(globalCtx->playerActorCsIds[8]));
+
+    pos = &player->actor.world.pos;
+    camera->player = NULL;
+
+    sp3C.x = ((180.0f * Math_SinS(this->unk_1E8E)) * Math_CosS(0xFA0)) + pos->x;
+    sp3C.y = (Math_SinS(0xFA0) * 180.0f) + pos->y;
+    sp3C.z = ((180.0f * Math_CosS(this->unk_1E8E)) * Math_CosS(0xFA0)) + pos->z;
+
+    camera->eye.x = ((sp3C.x - camera->eye.x) * arg2) + camera->eye.x;
+    camera->eye.y = ((sp3C.y - camera->eye.y) * arg2) + camera->eye.y;
+    camera->eye.z = ((sp3C.z - camera->eye.z) * arg2) + camera->eye.z;
+
+    camera->fov = ((this->unk_1E78 - camera->fov) * arg2) + camera->fov;
+    camera->at.y += 1.4444444f;
+}
+
+void func_80AF2654(EnTest7* this, GlobalContext* globalCtx, f32 arg2) {
+    Vec3f* pos;
+    Player* player = GET_PLAYER(globalCtx);
+    Camera* camera;
+    Vec3f sp30;
+
+    camera = Play_GetCamera(globalCtx, ActorCutscene_GetCurrentCamera(globalCtx->playerActorCsIds[8]));
+    camera->player = NULL;
+
+    pos = &player->actor.world.pos;
+
+    sp30.x = ((80.0f * Math_SinS(this->unk_1E8E)) * Math_CosS(0xBB8)) + pos->x;
+    sp30.y = (Math_SinS(0xBB8) * 80.0f) + pos->y;
+    sp30.z = ((80.0f * Math_CosS(this->unk_1E8E)) * Math_CosS(0xBB8)) + pos->z;
+
+    camera->eye.x = ((sp30.x - camera->eye.x) * arg2) + camera->eye.x;
+    camera->eye.y = ((sp30.y - camera->eye.y) * arg2) + camera->eye.y;
+    camera->eye.z = ((sp30.z - camera->eye.z) * arg2) + camera->eye.z;
+
+    camera->at.x = ((pos->x - camera->at.x) * arg2) + camera->at.x;
+    camera->at.y = (((pos->y + 40.0f) - camera->at.y) * arg2) + camera->at.y;
+    camera->at.z = ((pos->z - camera->at.z) * arg2) + camera->at.z;
+
+    camera->fov = ((this->unk_1E78 - camera->fov) * arg2) + camera->fov;
+}
+
+void func_80AF2808(EnTest7* this, GlobalContext* globalCtx, f32 arg2) {
+    Player* player = GET_PLAYER(globalCtx);
+
+    player->actor.shape.rot.y += 0x2EE0;
+    player->actor.scale.x = ((0.0f - this->unk_1E90) * arg2) + this->unk_1E90;
+    player->actor.scale.z = ((0.0f - this->unk_1E94) * arg2) + this->unk_1E94;
+}
+
+void func_80AF2854(EnTest7* this, GlobalContext* globalCtx) {
+    f32 temp;
+
+    if ((this->unk_1E54 >= 12) && (this->unk_1E54 < 31)) {
+        temp = (this->unk_1E54 - 12) / 18.0f;
+        func_80AF24D8(this, globalCtx, temp);
+    } else if ((this->unk_1E54 >= 79) && (this->unk_1E54 < 96)) {
+        temp = (this->unk_1E54 - 79) / 16.000000001f;
```

Comment:
```suggestion
        temp = (this->unk_1E54 - 79) / 16.0f;
```
If this doesn't work, does `/ 16`?

Reply:
Nope.